plugins/lsp/language-servers/default.nix

```
@@ -310,6 +310,16 @@ let
       name = "metals";
       description = "metals for Scala";
     }
+    {
+      name = "nextls";
+      description = "The language server for Elixir that just works.";
+      package = pkgs.next-ls;
+      serverName = "nextls";
+      cmd = cfg: [
+        (lib.getExe cfg.package)
```

```suggestion
        "nextls"
```
I think that this solution is more flexible.
Nixvim will ensure that `nextls` is on the `PATH`.

---

We don't do this for other LSPs. They all use an explicit absolute path into the nix store.

OTTOMH, I have a feeling the `cmd` is only used when `package` is non-null anyway?

---

For me, there is no link between the package and the cmd in most cases.
The `package` option (in nixvim) ensures that the LS package is set on the PATH within neovim.
The `cmd` tells neovim how to run the language server.

We could hardcode some `pkgs` path in `cmd` for extreme cases were the `PATH` mechanism doesn't work.
However, for me it should work here, no ?

---

> I have a feeling the cmd is only used when package is non-null anyway?

[nextls](https://github.com/neovim/nvim-lspconfig/blob/master/doc/server_configurations.md#nextls) does not set a `cmd` by default. I thought it would be a nice to have since it can be overwritten anyway.

---

> [nextls](https://github.com/neovim/nvim-lspconfig/blob/master/doc/server_configurations.md#nextls) does not set a `cmd` by default. I thought it would be a nice to have since it can be overwritten anyway.

This sure, but I would have set it to `["nextls" "--stdo"]`

---

So that it's completely independent from how the user has installed `nextls`.

---

> For me, there is no link between the package and the cmd in most cases. The `package` option (in nixvim) ensures that the LS package is set on the PATH within neovim. The `cmd` tells neovim how to run the language server.

Ok that makes sense.

Unless I'm mis-remembering, most (all?) cmds currently use an absolute path, so should we be going through and changing that in another PR?